src/vendorpromo/views.py

```diff
@@ -105,6 +107,158 @@ class AffiliateDeleteView(LoginRequiredMixin, DeleteView):
     success_url = reverse_lazy('affiliate-list')
 
 
+def create_promo_offer(promo_campaign, products, cost):
+    now = timezone.now()
+    promo_offer = Offer()
+    promo_offer.name = promo_campaign.name
+    promo_offer.start_date = now
+    promo_offer.site = promo_campaign.site
+    promo_offer.is_promotional = True
+    promo_offer.save()
+
+    for product in products.all():
+        promo_offer.products.add(product)
```

Comment:
if products is already a queryset, you shouldn't need the .all() here to loop through them